### PR TITLE
Upgrade Feast transformer for v0.12.1

### PR DESCRIPTION
**Pull request by chinhuang007**
Made the following changes for Feast transformer to support the
latest Feast v0.12.1
1. Change the python client API to rest API
2. Use the new Feast feature server
3. Use python 3.8
4. Update sample outputs in doc

It is to replicate [this PR](https://github.com/kubeflow/kfserving/pull/1790) in KFServing

**What this PR does / why we need it**:
Update Feast transformer to work with the latest Feast version

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

1. Please confirm that if this PR changes any image versions, then that's the sole change this PR makes.

**Release note**:

```release-note

```

### CODE REVIEW

File: docs/samples/v1beta1/transformer/feast/setup.py [hunk 1]
```diff
@@ -22,19 +22,18 @@
 
 setup(
     name='driver_transformer',
-    version='0.1.0',
+    version='0.2.0',
     author_email='chhuang@us.ibm.com',
     license='../../LICENSE.txt',
-    url='https://github.com/kserve/kserve/docs/samples/v1beta1/transformer/feast/driver_transformer',
+    url='https://github.com/kubeflow/kfserving/docs/samples/v1beta1/fesat/driver_transformer',
```
Comment (pvaneck) [inline docs/samples/v1beta1/transformer/feast/setup.py:28]: Can we fix the path of this URL and using the kserve repo?

File: docs/samples/v1beta1/transformer/feast/driver_transformer/driver_transformer.py [hunk 2]
```diff
@@ -14,14 +14,14 @@
...
-from feast import Client
+logging.basicConfig(level=kfserving.constants.KFSERVING_LOGLEVEL)
 
-logging.basicConfig(level=kserve.constants.KSERVE_LOGLEVEL)
 
-
-class DriverTransformer(kserve.KFModel):
+class DriverTransformer(kfserving.KFModel):
     """ A class object for the data handling activities of driver ranking
     Task and returns a KFServing compatible response.
```
Comment (pvaneck) [inline docs/samples/v1beta1/transformer/feast/driver_transformer/driver_transformer.py:26]: s/KFServing/KServe/

File: docs/samples/v1beta1/transformer/feast/driver_transformer/driver_transformer.py [hunk 3]
```diff
@@ -14,14 +14,14 @@
 
 from typing import List, Dict
 import logging
-import kserve
+import kfserving
```
Comment (pvaneck) [inline docs/samples/v1beta1/transformer/feast/driver_transformer/driver_transformer.py:17]: `import kserve`

File: docs/samples/v1beta1/transformer/feast/driver_transformer/__main__.py [hunk 4]
```diff
@@ -13,15 +13,15 @@
 # limitations under the License.
 
 import argparse
-import kserve
+import kfserving
```
Comment (pvaneck) [inline docs/samples/v1beta1/transformer/feast/driver_transformer/__main__.py:16]: `import kserve`

File: docs/samples/v1beta1/transformer/feast/setup.py [hunk 5]
```diff
@@ -22,19 +22,18 @@
...
     description='Driver transformer',
     long_description=open('README.md').read(),
     python_requires='>=3.6',
     packages=find_packages("driver_transformer"),
     install_requires=[
-        "kfserving>=0.5.1",
+        "kfserving>=0.6.0",
```
Comment (pvaneck) [inline docs/samples/v1beta1/transformer/feast/setup.py:34]: Can use kserve>=0.7.0rc0

**Review by pvaneck — COMMENTED** [review]
Mind doing a KFServing -> KServe find and replace in the docs and comments for this transformer?

**Comment by yuzisun** [comment]
@chinhuang007 I have merged your original commit from kfserving to kserve, please help double check if the all the necessary changes are there ?

**Comment by chinhuang007** [comment]
@yuzisun Just verified with the latest in https://github.com/kserve/kserve and everything looks good.

**Comment by yuzisun** [comment]
> @yuzisun Just verified with the latest in https://github.com/kserve/kserve and everything looks good.

Awesome! we can close this one then

**Comment by chinhuang007** [comment]
Sounds good.

**Pull request closed without merging**